### PR TITLE
Auth middleware

### PR DESCRIPTION
**Pull request by Selvyfitriani**
Provide an endpoint for getting user info with custom jwt validation as middleware.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,10 +6,13 @@ tmp
...
+.env.testing
 
 # The development sqlite file
 database/*.sqlite
 
 package-lock.json
 
 jwt.json
+
+test.json
```
Comment (ayocodingit) [inline .gitignore:18]: please save to env, can use base64 encryption, later decrypt to use it,

Comment (tia3nt) [reply .gitignore:18]: ok Mas

**Comment by tia3nt** [comment]
Mas Firman, Is it Ok if I keep the username and password separated from the Config file, because I think, username and password may varied in a real testcase.
Is it also possible if i parse the url so that the config file more flexible and reusable, I mean like 
define the realms name, scope name, end point first then i can define complete url separately.
because as far as I notice, token url, jwks url are basically only difference at the end point part.
I was trying a few times but no luck.

**Comment by ayocodingit** [comment]
> Mas Firman, Is it Ok if I keep the username and password separated from the Config file, because I think, username and password may varied in a real testcase. Is it also possible if i parse the url so that the config file more flexible and reusable, I mean like define the realms name, scope name, end point first then i can define complete url separately. because as far as I notice, token url, jwks url are basically only difference at the end point part. I was trying a few times but no luck.

Yes, just make it as needed, but save it in the env.

**Comment by tia3nt** [comment]
o Ok, thank you for your input Mas

File: package.json [hunk 2]
```diff
@@ -23,9 +23,11 @@
     "@adonisjs/fold": "^4.0.9",
     "@adonisjs/framework": "^5.0.9",
     "@adonisjs/ignitor": "^2.0.8",
-    "@adonisjs/lucid": "^6.1.3"
+    "@adonisjs/lucid": "^6.1.3",
+    "@adonisjs/session": "^1.1.0",
```
Comment (ayocodingit) [inline package.json:27]: please remove this package because unused

File: app/Middleware/Auth.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  async handle ({ request, response }, next ) {
+    const token = await this.getToken(request)
+    const key = await this.getKey()
+
+    try {
+      const decoded = jwt.verify(token, key)
+      const user = {name: decoded['name'], email: decoded['email']}
+      request.user = user
+
+      await next()
```
Comment (ayocodingit) [inline app/Middleware/Auth.js:45]: please remove await next(), just  method next()

File: test/functional/user-controller.spec.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  const response = await client
+    .get('/me')
+    .header({
+      Authorization: `Bearer ${token}`,
+    })
+    .end()
+  
+  response.assertStatus(200)
+  response.assertJSONSubset({
+    name: 'Selvy Fitriani',
```
Comment (ayocodingit) [inline test/functional/user-controller.spec.js:39]: move name to env,

**Review by ayocodingit — CHANGES_REQUESTED** [review]

File: app/Middleware/Auth.js (hunk 3 above)
Comment (tia3nt) [reply app/Middleware/Auth.js:45]: Ok Mas

**Review by ayocodingit — APPROVED** [review]

**Pull request merged**